cognite/extractorutils/statestore.py

```diff
@@ -188,6 +178,36 @@ def synchronize(self) -> None:
         """
         pass
 
+
+class AbstractStateStore(_BaseStateStore, ABC):
+    """
+    Base class for a state store.
+
+    Args:
+        save_interval: Automatically trigger synchronize each m seconds when run as a thread (use start/stop
+            methods).
+        trigger_log_level: Log level to log synchronize triggers to.
+        thread_name: Thread name of synchronize thread.
+        cancellation_token: Token to cancel event from elsewhere. Cancelled when stop is called.
+    """
+
+    def __init__(
+        self,
+        save_interval: Optional[int] = None,
+        trigger_log_level: str = "DEBUG",
+        thread_name: Optional[str] = None,
+        cancellation_token: Optional[CancellationToken] = None,
+    ):
+        super().__init__(
+            save_interval=save_interval,
+            trigger_log_level=trigger_log_level,
+            thread_name=thread_name,
+            cancellation_token=cancellation_token,
+        )
+
+        self._local_state: Dict[str, Dict[str, Any]] = {}
```

Comment:
Is it feasible to use a type union here? I see the dictionary key "digest" being used in the AbstractHashStateStore, might be good to have typing.

Reply:
Not without a breaking change.